server/src/main/java/org/spine3/server/command/CommandBus.java

```diff
@@ -124,27 +127,37 @@ public void unregister(CommandHandler handler) {
      *
      * <p>If {@code Command} instance is passed, its message is extracted and validated.
      *
-     * @param command the command message to check
+     * @param commandOrMessage the command message to check
      * @return the result of {@link Responses#ok()} if the command is supported,
      *         {@link CommandValidation#unsupportedCommand(Message)} otherwise
      */
-    public Response validate(Message command) {
-        checkNotNull(command);
-        final CommandClass commandClass = CommandClass.of(command);
+    public Response validate(Message commandOrMessage) {
+        checkNotNull(commandOrMessage);
 
-        if (dispatcherRegistered(commandClass)) {
-            return Responses.ok();
+        final Message message = toMessage(commandOrMessage);
+        final CommandClass commandClass = CommandClass.of(message);
+        if (isUnsupportedCommand(commandClass)) {
+            return CommandValidation.unsupportedCommand(message);
         }
-
-        if (handlerRegistered(commandClass)) {
-            return Responses.ok();
+        final MessageValidator validator = new MessageValidator(CommandFieldValidatorFactory.newInstance());
+        validator.validate(message);
+        if (validator.isMessageInvalid()) {
```

Comment:
But we have an inspection `Boolean method is always inverted`. And client code needs to know if the message is invalid. So I added `isValid` method and inverted the variable.